### PR TITLE
force setup.py version bump in PRs

### PR DESCRIPTION
**Pull request by maorfr**
follows up on #2326

this PR will force each future PR to bump the version in setup.py to a higher version.

### CODE REVIEW

**Comment by maorfr** [comment]
the approach of adding this test as a test and not a make target or a bash script was inspired by #652

File: reconcile/utils/semver_helper.py [hunk 1]
```diff
@@ -21,3 +21,7 @@ def sort_versions(versions: Iterable[str]) -> List[str]:
     """
     semver_versions = sorted([parse_semver(v) for v in versions])
     return [str(v) for v in semver_versions]
+
+
+def is_version_bumped(current_version: str, previous_version: str) -> bool:
```
Comment (fishi0x01) [inline reconcile/utils/semver_helper.py:26]: Where is this used?

**Comment by maorfr** [comment]
depends on https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/36837

**Review by fishi0x01 — COMMENTED** [review]
Lets highlight that this is a hard requirement now: 

Each PR **must** ... CI tests will fail if version is not bumped.

**EDIT** Ref to code passage wasnt linked :thinking: This should refer to README update

**Comment by maorfr** [comment]
@fishi0x01 done

**Comment by fishi0x01** [comment]
Nice :+1: 
Probably also worth a mention in team chat as this will affect everybody from now on - pretty sure this might lead to confusion initially

**Review by fishi0x01 — APPROVED** [review]

**Comment by maorfr** [comment]
release to pypi was a success: https://pypi.org/project/qontract-reconcile/0.5.0/

**Pull request merged**